modules/kernels/integration.py

```
@@ -0,0 +1,342 @@
+        p.grad = None
+
+    out_fused = block(x)
+    out_fused.backward(grad)
+    grads_fused = {n: p.grad.clone() for n, p in block.named_parameters() if p.grad is not None}
+
+    max_w_diff = max(
+        (grads_fused[n] - grads_ref[n]).abs().max().item()
+        for n in grads_ref
+    )
+    print(f"Block weight grad max diff: {max_w_diff:.4e}")
+    print(f"\nIntegration works! Use model.eval() for ONNX export fallback.")
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)